src/main/java/com/renatusnetwork/momentum/data/races/gamemode/RaceRequest.java

```diff
@@ -139,8 +139,12 @@ else if (validateStats.inLevel() && validateStats.getLevel().isElytra())
         return null;
     }
 
-    public boolean equals(PlayerStats sender, PlayerStats requester)
+    public boolean equals(PlayerStats sender, PlayerStats requested)
     {
-        return (sender.equals(getSender()) && requester.equals(getRequested())) || (sender.equals(getRequested()) && requester.equals(getSender()));
+        return (sender.equals(getSender()) && requested.equals(getRequested())) || (sender.equals(getRequested()) && requested.equals(getSender()));
+    }
+
+    public boolean isAcceptableRequest(PlayerStats sender, PlayerStats requested) {
+        return sender.equals(getRequested()) && requested.equals(getSender());
```

Comment:
is there any way we could make this more clear so that sender.equals(getSender()) instead of opposites?

Reply:
without actually swapping the parameters, we have to check if the player sending the /race accept cmd is actually the one who got requested instead of vice versa. changing the equals method would mess with the accept timeout since its sent with the requester as the sender whereas accepting the request sets the requested as the sender which is the condition need to accept.